pkg/controller/kfservice/resources/knative/service_test.go

```diff
@@ -479,3 +479,198 @@ func TestKFServiceToKnativeService(t *testing.T) {
+				Container: v1.Container{
+					Image: "transformer:v2",
+				},
+			},
+		},
+		Predictor: v1alpha2.PredictorSpec{
+			DeploymentSpec: v1alpha2.DeploymentSpec{
+				MinReplicas:        1,
+				MaxReplicas:        3,
+				ServiceAccountName: "testsvcacc",
+			},
+			Tensorflow: &v1alpha2.TensorflowSpec{
+				StorageURI:     "s3://test/mnist-2/export",
+				RuntimeVersion: "1.13.0",
+			},
+		},
+	}
+
+	var defaultService = &knservingv1alpha1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.DefaultTransformerServiceName("mnist"),
+			Namespace: "default",
+		},
+		Spec: knservingv1alpha1.ServiceSpec{
+			ConfigurationSpec: knservingv1alpha1.ConfigurationSpec{
+				Template: &knservingv1alpha1.RevisionTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{"serving.kubeflow.org/kfservice": "mnist"},
+						Annotations: map[string]string{
+							"autoscaling.knative.dev/class":    "kpa.autoscaling.knative.dev",
+							"autoscaling.knative.dev/target":   "1",
+							"autoscaling.knative.dev/minScale": "1",
+							"autoscaling.knative.dev/maxScale": "3",
+						},
+					},
+					Spec: knservingv1alpha1.RevisionSpec{
+						RevisionSpec: v1beta1.RevisionSpec{
+							TimeoutSeconds: &constants.DefaultTimeout,
+							PodSpec: v1.PodSpec{
+								ServiceAccountName: "testsvcacc",
+								Containers: []v1.Container{
+									{
+										Image: "transformer:latest",
+										Args: []string{
+											constants.ModelServerArgsModelName,
+											kfsvc.Name,
+											constants.ModelServerArgsPredictorHost,
+											constants.DefaultPredictorServiceName(kfsvc.Name),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	var canaryService = &knservingv1alpha1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.CanaryTransformerServiceName("mnist"),
+			Namespace: "default",
+		},
+		Spec: knservingv1alpha1.ServiceSpec{
+			ConfigurationSpec: knservingv1alpha1.ConfigurationSpec{
+				Template: &knservingv1alpha1.RevisionTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{"serving.kubeflow.org/kfservice": "mnist"},
+						Annotations: map[string]string{
+							"autoscaling.knative.dev/class":    "kpa.autoscaling.knative.dev",
+							"autoscaling.knative.dev/target":   "1",
+							"autoscaling.knative.dev/minScale": "2",
+							"autoscaling.knative.dev/maxScale": "4",
+						},
+					},
+					Spec: knservingv1alpha1.RevisionSpec{
+						RevisionSpec: v1beta1.RevisionSpec{
+							TimeoutSeconds: &constants.DefaultTimeout,
+							PodSpec: v1.PodSpec{
+								ServiceAccountName: "testsvcacc",
+								Containers: []v1.Container{
+									{
+										Image: "transformer:v2",
+										Args: []string{
+											constants.ModelServerArgsModelName,
+											kfsvc.Name,
+											constants.ModelServerArgsPredictorHost,
+											constants.CanaryPredictorServiceName(kfsvc.Name),
```

Comment:
The py Transformer class expects a protocol string as an argument - should this code be setting it?

Reply:
it is defaulting to tensorflow protocol in kfserver, on kfservice spec we have not exposed the protocol anywhere.